caikit/core/model_management/local_job_predictor.py

```diff
@@ -0,0 +1,221 @@
+            self._prediction_func_name = prediction_func_name
+            self._result_type = None
+            super().__init__(*args, extra_path_args=[RESULT_FILE_NAME], **kwargs)
+
+        def run(self, *args, **kwargs):
+            """Run the prediction and save the results in a binary format to a file"""
+
+            # Create parent prediction directory before starting request
+            save_path_pathlib = Path(self.save_path)
+            log.debug3(
+                "Creating prediction result directory %s", save_path_pathlib.parent
+            )
+            save_path_pathlib.parent.mkdir(exist_ok=True, parents=True)
+
+            with alog.ContextTimer(
+                log.debug, "Inference in job %s finished in: ", self.id
+            ):
+                model_run_fn = getattr(self._model_instance, self._prediction_func_name)
+                infer_result = model_run_fn(*args, **kwargs)
+                self._result_type = infer_result.__class__
+
+            # If save path was provided then output result
+            if self.save_path is not None:
+                self._save_result(infer_result)
+
+            self._completion_time = self._completion_time or datetime.now()
+            log.debug2("Completion time for %s: %s", self.id, self._completion_time)
+            return infer_result
+
+        def result(self) -> DataObjectBase:
+            """Fetch the result from the result directory"""
+            # Wait for future to complete
+            self.wait()
+
+            result_path = Path(self.save_path)
+            if not result_path.exists():
+                raise CaikitCoreException(
+                    CaikitCoreStatusCode.NOT_FOUND,
+                    f"Prediction result for {self.id} is not found",
+                )
+
+            assert self._result_type
+            return self._result_type.from_binary_buffer(result_path.read_bytes())
+
+        def _delete_result(self):
+            # Delete the result.bin if one exists
+            super()._delete_result()
+
+            # Delete parent directory to clear out future id
+            if self.save_path:
+                parent_path = Path(self.save_path).parent
+                if parent_path.exists():
+                    shutil.rmtree(parent_path, ignore_errors=True)
+
+        def _save_result(self, result: DataObjectBase):
+            """Helper function to save a result to disk. This abstraction helps
+            subclasses implement their own save method like S3"""
+            save_path_pathlib = Path(self.save_path)
+            log.debug("Saving inference %s to %s", self.id, self.save_path)
+            save_path_pathlib.parent.mkdir(exist_ok=True, parents=True)
+            with alog.ContextTimer(
+                log.debug, "Inference %s saved in: ", self.id
+            ) and save_path_pathlib.open("wb") as output_file:
+                output_file.write(result.to_binary_buffer())
+
+    LocalModelFuture = LocalJobPredictorFuture
```

Comment:
This seems misplaced here? I think `LocalModelFuture` should only be in the trainer?

Reply:
Yep! Darn copy/pasta